google/api_core/retry.py

```diff
@@ -139,15 +139,15 @@ def exponential_sleep_generator(initial, maximum, multiplier=_DEFAULT_DELAY_MULT
     Yields:
         float: successive sleep intervals.
     """
-    delay = initial
+    delay = min(initial, maximum)
```

Comment:
Are they never negative ?

Reply:
Who are they? initial and maximum? Yes, they are. There is still a chance that a user will set the values as negative, but it is very unlikely and could be done only either by mistake or with intention to break it all. We could have validated that the values are not negative but seems like gapic libraries do not perform these kinds of validation (i.e. by convention we don't go as far as validating ranges of all integer params), so not validating it here (this is not changed in this PR and has been like this before)